src/SfxWeb/src/app/modules/upgrade-progress/health-policy-check/health-policy-check.component.ts

```diff
@@ -0,0 +1,105 @@
+
+    let minDurationLeft: string = "";
+    let middlePhase = `Stable Duration Check  - ${healthCheckStableDuration}`;
+    let middleTooltip = `The upgrade must be healthy as defined by the monitoring policies for ${healthCheckStableDuration} before completing the health policy check.`
+    this.healthCheckPhaseText = "Stable duration check will start after the wait duration completes.";
+
+    this.healthCheckDurationLeft = TimeUtils.getDurationMilliseconds(this.healthCheckPhaseDuration);
+    this.failed = false;
+
+    if (this.healthCheckPhase === "WaitDuration") {
+      let durationLeft = TimeUtils.getDurationMilliseconds(this.monitoringPolicy.HealthCheckWaitDurationInMilliseconds) - this.healthCheckDurationLeft;;
+      durationLeft += TimeUtils.getDurationMilliseconds(this.monitoringPolicy.HealthCheckStableDurationInMilliseconds);
+      minDurationLeft = TimeUtils.getDuration(durationLeft);
+
+      this.HealthCheckDurationOverall = TimeUtils.getDurationMilliseconds(this.monitoringPolicy.HealthCheckWaitDurationInMilliseconds);
+      this.displayBottomText = "Wait Time Duration Left";
+      this.displayTopText = "Wait Time Elapsed";
+      this.color = 'var(--accent-darkblue)';
+      this.currentPhaseIndex = 1;
+
+    } else if (this.healthCheckPhase === "Retry") {
+      this.currentPhaseIndex = 2;
+      minDurationLeft = TimeUtils.getDuration(this.monitoringPolicy.HealthCheckStableDurationInMilliseconds) + " once stable";
+      middlePhase = `Retry Duration Check  - ${healthCheckRetryTimeout}`;
+      this.healthCheckPhaseText = "If the health policy becomes healthy, the retry check will move to stable and move towards success.";
+      middleTooltip = `The upgrade has ${healthCheckRetryTimeout} to become healthy as defined by the monitoring policies or the upgrade domain will fail. Adjust your cluster health policy if required.`
+
+      this.HealthCheckDurationOverall = TimeUtils.getDurationMilliseconds(this.monitoringPolicy.HealthCheckRetryTimeoutInMilliseconds);
+      this.displayBottomText = "Retry Time Duration Left";
+      this.displayTopText = "Retry Time out Elapsed";
+      this.color = 'var(--badge-error)';
+      this.failed = true;
+    } else if (this.healthCheckPhase === "StableDuration") {
+      this.currentPhaseIndex = 2;
+      minDurationLeft = TimeUtils.getDuration(TimeUtils.getDurationMilliseconds(this.monitoringPolicy.HealthCheckStableDurationInMilliseconds) - this.healthCheckDurationLeft);
+      this.healthCheckPhaseText = "If the health policy becomes unhealthy, the stable check will move to retry and potentially fail.";
+      this.HealthCheckDurationOverall = TimeUtils.getDurationMilliseconds(this.monitoringPolicy.HealthCheckStableDurationInMilliseconds);
+      this.displayBottomText = "Stable Time Duration Left";
+      this.displayTopText = "Stable Time Elapsed";
+      this.color = 'green';
```

Comment:
any reason not to use `--badge-ok` for green here?

Reply:
added css var to start moving to "proper" green